tests/testthat/test-merge_sce_list.R

```diff
@@ -397,59 +469,82 @@ test_that("merging SCEs with same altExp features works as expected, with altexp
     unname()
   expect_equal(colnames(merged_altexp), expected_colnames)
 
-})
+  # Check colData columns
+  expected_coldata <- c(
+    "sum",
+    "detected",
+    "altexps_adt_sum",
+    "altexps_adt_detected",
+    "altexps_adt_percent",
+    batch_column,
+    "cell_id"
+  )
 
+  expect_true(
+    setequal(
+      colnames(colData(merged_sce)),
+      expected_coldata
+    )
+  )
+})
 
 
 
-test_that("merging SCEs with different altExp features works as expected, with altexps", {
 
+test_that("merging SCEs with 1 altexp but different features fails as expected, with altexps", {
   # keep only the first 3 features from the first SCE
-  altExp(sce_list_with_altexp[[1]]) <- altExp(sce_list_with_altexp[[1]])[1:3]
+  altExp(sce_list_with_altexp[[1]]) <- altExp(sce_list_with_altexp[[1]])[1:3, ]
 
 
-  merged_sce <- merge_sce_list(
-    sce_list_with_altexp,
-    batch_column = batch_column,
-    # "total" should get removed
-    retain_coldata_cols = retain_coldata_cols,
-    # this row name should not be modified:
-    preserve_rowdata_cols = c("gene_names")
+  expect_error(
+    merge_sce_list(
+      sce_list_with_altexp,
+      batch_column = batch_column,
+      # "total" should get removed
+      retain_coldata_cols = retain_coldata_cols,
+      # this row name should not be modified:
+      preserve_rowdata_cols = c("gene_names")
+    )
   )
+})
 
-  merged_altexp <- altExp(merged_sce)
-
-
-  expect_true(altExpNames(merged_sce) == altexp_name)
-  expect_equal(dim(merged_altexp), c(num_altexp_features, total_cells))
-  expect_equal(rownames(merged_altexp), full_altexp_features)
 
-  expected_colnames <- sce_list_with_altexp |>
-    purrr::imap(
-      \(sce, sce_name) glue::glue("{sce_name}-{colnames(sce)}")
-    ) |>
-    unlist() |>
-    unname()
-  expect_equal(colnames(merged_altexp), expected_colnames)
 
-})
 
+test_that("merging SCEs where 1 altExp is missing works as expected, with altexps", {
+  sce_list_with_altexp[["sce4"]] <- removeAltExps(sce_list_with_altexp[[1]])
```

Comment:
Oh, I meant to suggest a different (even easier!) approach here, because this leaves in the altExp stats in the colData. 
```suggestion
  sce_list_with_altexp[["sce4"]] <- sce_list[[1]]
```